test_ffi/tests/integration_tests.rs

```diff
@@ -105,6 +109,14 @@ fn basic() {
     buf: [1, 2, 3, 4, 5, 6, 7, 8]\n\
     logCallback\n\
     30\n\
+    255 65535 4294967295 4294967296 123.456 789.876 -1 -2 -3 -4 -1000 1000 12345.67891 12345.679 12345.67891 12345.679 12345.67891 12345.679 12345.67891\n\
+    255 65535 4294967295 4294967296 123.456 789.876 -1 -2 -3 -4 -1000 1000 12345.67891 12345.679 12345.67891 12345.679 12345.67891 12345.679 12345.67891\n\
+    0\n\
+    0\n\
+    0\n\
+    0\n\
+    78\n\
+    45\n\
```

Comment:
Output in the test is 78 on this row as well. Is this correct?

Reply:
This is the consequence of: https://github.com/denoland/deno/blob/73f05c793342870cb88afd3006652f304d3c2d00/test_ffi/tests/test.js#L483-L488

The output of the test should be 78. The problem is that on `aarch64-apple-darwin` the test outputs 45 due to the aforementioned V8 bug. 

The bug has already been reported in https://bugs.chromium.org/p/v8/issues/detail?id=13171. 

Considering that this test will unequivocally detect when a new version of V8 fixes the bug I am now slightly leaning towards opinion 2. However, CI does not run tests on `aarch64-apple-darwin`...

Opinions?

Reply:
Hmm... If the adaptation is easy to do and easy to remove, then I guess option 2 would not be bad ... Two wrongs do make a right sometimes.

Still, I wouldn't really mind if choice 1 was made either. That would put pressure on V8 to fix their stuff, as I'm sure a lot of people would come complaining to Deno repo about FFI being broken on their Macs and we could just guide them to the V8 issue to voice their displeasure.

Reply:
> ... and we could just guide them to the V8 issue to voice their displeasure.

And yet, maybe that wouldn't be the most respectful way of treating their nug tracking forums? :)

How about we adapt, so code *works* today, and have a "TODO" comment with the code to be exchanged for after the V8 issue is fixed?

Reply:
Ok let's go with 1 & do 2. in a follow up? I don't want to block this PR any further; the goal right now is to move away from tinycc.

Reply:
>  lot of people would come complaining to Deno repo about FFI being broken on their Macs and we could just guide them to the V8 issue to voice their displeasure.

This is true, and by default I would chose this approach. But I fear the solution on V8 side might not be as straightforward as once anticipated (according to @devsnek), and I intuitively think option 2 is actually easier than option 1 (I would argue against removing the test, and we cannot leave a broken test for the folks with M1s, so we are left with conditional "expected failure", somehow). Also, it always kinda feels a bit like using the users (and their Deno DX) as pawns. After all, for many the story usually ends at "Deno FFI is broken in my Mac".

Reply:
Implementing V8 calling convention is trivial:
```diff
-        let size_original = param.size();
+        // let size_original = param.size();
+        let size_original = 8;
```
I just need to clarify if V8 expects 8/16 bit stack parameters to be extended, considering they are 8-byte aligned. I won't have time till tonight to get to it, so if anybody else wants to take the lead, be my gest!

The more laborious part of adapting to V8 calling convention would had been in `Aarch64Apple::allocate_stack()`, but funny story, I had it wrong to begin with 🤦 (the kind of wrong that matches V8 expectations)

Reply:
pushed changes for option 2: adapt to V8 calling convention.